src/draw/nxp/pxp/lv_pxp_osa.c

```diff
@@ -83,26 +88,23 @@ static pxp_cfg_t _pxp_default_cfg = {
 
 void PXP_IRQHandler(void)
 {
-#if defined(SDK_OS_FREE_RTOS)
-    BaseType_t xHigherPriorityTaskWoken = pdFALSE;
-#endif
-
     if(kPXP_CompleteFlag & PXP_GetStatusFlags(PXP_ID)) {
         PXP_ClearStatusFlags(PXP_ID, kPXP_CompleteFlag);
-#if defined(SDK_OS_FREE_RTOS)
-        xSemaphoreGiveFromISR(xPXPIdleSemaphore, &xHigherPriorityTaskWoken);
-
-        /* If xHigherPriorityTaskWoken is now set to pdTRUE then a context switch
-        should be performed to ensure the interrupt returns directly to the highest
-        priority task.  The macro used for this purpose is dependent on the port in
-        use and may be called portEND_SWITCHING_ISR(). */
-        portYIELD_FROM_ISR(xHigherPriorityTaskWoken);
+#if LV_USE_OS
+        lv_thread_sync_signal(&pxp_sync);
```

Comment:
The is still one issue:
`lv_thread_sync_signal()` was not ment to work work from ISR.

The implementation from ISR should be using FreeRTOS `vTaskNotifyGiveFromISR()` instead of `xTaskNotifyGive()`.

Reply:
We should add a new osal API in FreeRTOS (and same for Zephyr) like `lv_thread_sync_sygnal_from_isr()`.

Reply:
I added this method to the OSAL, however I would need help from experts of all other plattforms if I have done it correctly.

Reply:
The API does not look good on FreeRTOS either.
1. You should use `vTaskNotifyGiveFromISR()`
2. From documentation:
```
vTaskNotifyGiveFromISR() will set *pxHigherPriorityTaskWoken to pdTRUE if sending the notification caused a task to unblock, and the unblocked task has a priority higher than the currently running task.

If vTaskNotifyGiveFromISR() sets this value to pdTRUE then a context switch should be requested before the interrupt is exited. See the example below. 
```


```
/* The transmit end interrupt. */
void vTransmitEndISR( void )
{
BaseType_t xHigherPriorityTaskWoken = pdFALSE;

    /* At this point xTaskToNotify should not be NULL as a transmission was
    in progress. */
    configASSERT( xTaskToNotify != NULL );

    /* Notify the task that the transmission is complete. */
    vTaskNotifyGiveIndexedFromISR( xTaskToNotify, 0, &xHigherPriorityTaskWoken );

    /* There are no transmissions in progress, so no tasks to notify. */
    xTaskToNotify = NULL;

    /* If xHigherPriorityTaskWoken is now set to pdTRUE then a context switch
    should be performed to ensure the interrupt returns directly to the highest
    priority task.  The macro used for this purpose is dependent on the port in
    use and may be called portEND_SWITCHING_ISR(). */
    portYIELD_FROM_ISR( xHigherPriorityTaskWoken );
}
```